beamer/tests/agent/multiple_processors/test_multiple_event_processors.py

```diff
@@ -0,0 +1,224 @@
+import contextlib
+import json
+import os
+import signal
+import time
+from pathlib import Path
+from subprocess import DEVNULL, PIPE
+
+import ape
+import eth_account
+import psutil
+import yaml
+from eth_utils import to_canonical_address
+from yaml.loader import SafeLoader
+
+from beamer.agent.agent import Agent
+from beamer.agent.config import Config
+from beamer.agent.contracts import ContractInfo
+from beamer.agent.typing import BlockNumber, ChainId
+from beamer.agent.util import TokenChecker, make_web3, transact
+from beamer.tests.conftest import Contracts
+from beamer.tests.util import Sleeper
+
+SLAVE_TEST_PATH = Path(__file__).parent / "_test_slave.py"
+CONFIG_PATH = ape.project.local_project.path / ape.project.local_project.config_file_name
+
+
+# ape local account, to be used for fulfilling requests.
+# The private key here corresponds to the 10th account ganache creates on
+# startup.
+_LOCAL_ACCOUNT = ape.accounts.test_accounts[-1]
+
+
+def _start_ganache(chain_id: ChainId, port: int) -> psutil.Popen:
+    cmd = [
+        "ganache-cli",
+        "--chain.vmErrorsOnRPCResponse",
+        "true",
+        "--server.port",
+        str(port),
+        "--miner.blockGasLimit",
+        "12000000",
+        "--wallet.totalAccounts",
+        "10",
+        "--hardfork",
+        "istanbul",
+        "--wallet.mnemonic",
+        "brownie",
+        "--chain.chainId",
+        str(chain_id),
+    ]
+    return psutil.Popen(cmd, stdin=DEVNULL, stdout=PIPE, stderr=PIPE)
+
+
+@contextlib.contextmanager
+def _configure_ganache_port(port: int):
+    try:
+        with open(CONFIG_PATH) as _f:
+            current_config = yaml.load(_f, Loader=SafeLoader)
+        current_port = current_config["ganache"]["server"]["port"]
+        current_config["ganache"]["server"]["port"] = port
+        with open(CONFIG_PATH, "w") as _f:
+            yaml.dump(current_config, _f, sort_keys=False, default_flow_style=False)
```

Comment:
This part should be moved outside the `try` block. We only want to do the revert to the original config if
we replaced it _and_ an exception happened during `yield`.

Also, please rename `_f` to just `f` or `fp` (think we have that elsewhere) -- I don't see a reason for the underscore here.